plugins/module_utils/user.py

```diff
@@ -359,9 +359,18 @@ def user_mod(cursor, user, host, host_all, password, encrypted,
                     revoke_privs = list(set(new_priv[db_table]) & set(curr_priv[db_table]))
                 else:
                     # When replacing (neither append_privs nor subtract_privs), grant all missing privileges
-                    # and revoke existing privileges that were not requested.
+                    # and revoke existing privileges that were not requested...
                     grant_privs = list(set(new_priv[db_table]) - set(curr_priv[db_table]))
                     revoke_privs = list(set(curr_priv[db_table]) - set(new_priv[db_table]))
+
+                    # ... but do not revoke GRANT option when it's already allowed
+                    # and already in privs.
+                    #
+                    # For more details
+                    # https://github.com/ansible-collections/community.mysql/issues/77#issuecomment-1209693807
+                    if 'GRANT' in new_priv[db_table] and 'GRANT' in curr_priv[db_table]:
+                        grant_privs.append('GRANT')
```

Comment:
The comment does not fit to the code, "do not revoke" would be rather `revoke_privs.remove('GRANT')` -- but that would be redundant, because `revoke_privs = list(set(curr_priv[db_table]) - set(new_priv[db_table]))` already covers it.